modules/mcuboot/CMakeLists.txt

```diff
@@ -231,6 +231,16 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
       )
   endif()
 
+  # Enable the same option in the MCUBoot child image
+  if (CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION)
+    add_overlay_config(
+      mcuboot
+      ${ZEPHYR_NRF_MODULE_DIR}/modules/mcuboot/hw_counters.conf
+    )
+
+    set(imgtool_security_counter --security-counter ${CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_VALUE})
```

Comment:
This is totally decoupled from the version number correct?
So if I as a customer want to make an update I'll have to manually increment this number?

Reply:
That is my understanding.

I see from the docs that the vendor could even sign an image with a lower firmware version and the same or higher security counter value.

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/mcuboot/design.html#hardware-based-downgrade-prevention

Reply:
Seems a bit weird but I won't block on this thanks for the clarification.